### DISCUSSION TITLE
Using a Knob with SDRpp

### DISCUSSION THREAD

**Discussion by ja-Bertolin** [AlexandreRouma/SDRPlusPlus#559] [Q&A]
Hi

I am trying to use the gadget: https://sdrgadgets.co.uk/universal/39-universal-vfo-tuner-plus.html with SDRpp and, even though it works fine with GQRX, when I am trying to tune a frequency the accuracy of the tuner is not so smooth, jump several Khz not 1 by 1.

Could it be related to the hardware or the app? With GQRX works fine. Is there anything that I should do to use it with this app?

I am using: MacOS  BigSur with AirSpyHf+

Thanks

**Comment by AlexandreRouma** [comment]
Hello,

I have no idea how that thing works so not much I can say :/
I haven't even heared of something like that until now.
If it's supposed to act as a mouse scrollwheel, the mouse cursor has to be in the waterfall to work.
And if the keypad is a normal keyboard, the mouse cursor has to hover the frequency input digits

**Comment by ja-Bertolin** [comment]
Hello Alexandre, thanks for reply

The unit works like as follows:
- It is like a mouse scrollwheel as you mention. Only works if the cursor of the mouse is placed in the waterfall or spectrum or on top of the numbers of the frequency
- The keyboards works fine,using the numbers it changes the frequency
- If you are on AM the change of the frequency is 1Khz, in USB it is 0.1Khz

The problem is the same as with my mouse, when scrolling frequency or takes a while to change or changes too fast but not smoothly. How could I control the speed of the mouse scrollwheel with sdrpp?

Thanks

**Reply by AlexandreRouma** [reply]
the frequency change of each scroll-wheel "tick" is governed by the "Snap interval" parameter. It can be configured to something different for each type of demodulator

**Reply by ja-Bertolin** [reply]
Hello Alexandre, 

I think that it will be better to move this question to Issues as the ticks governed by Snap Internal do not correspond to the movement of the knob as it does for instance in GQRX.

Thanks

**Comment by ja-Bertolin** [comment]
Hi @AlexandreRouma , I have been testing other SDR software (such as CubicSDR) and also it works fine with the mouse scroll-wheel. However, with SDR++ independently of Snap Interval value the frequency change is not linear: for every scroll-wheel tick one step. I do not know whether it happens the same with Windows but at least with Mac it does not work fine, only with GQRX or CubicSDR

I have been reviewing the configuration of the mouse in System Preferences but there is nothing that could help in such way.

Any of the other users have experienced the same behaviour?
